### DISCUSSION TITLE
OSU Hangs Randomly on NVIDIA 555 - NixOS Wayland

### DISCUSSION THREAD

**Discussion by A1029384756** [ppy/osu#28378] [Q&A]
### Type

Crash to desktop

### Bug description

Hang followed by a crash to desktop randomly when running 555 beta drivers (not present in 550). Reproducible in both the `flatpak` and `nixpkgs` version of the Lazer client.

### Screenshots or videos

https://github.com/ppy/osu/assets/35206453/23d8875b-1b3a-4212-a0ba-95912fe76c57

### Version

2024.521.2

### Logs

[logs.zip](https://github.com/user-attachments/files/15523369/logs.zip)

**Comment by smoogipoo** [comment]
Graphics driver issues are out of our control.

**Comment by shrublet** [comment]
Something similar happens on Fedora + GNOME too ever since the 555 driver update (screen freezes instead, but game doesn't crash and audio still plays). Firefox also currently has issues with these drivers as they're the first to implement explicit sync stuff, so a lot of Wayland apps are not playing too nicely. Not a total solution, but for now, disabling Wayland (in flatseal for flatpak) fixes the freezing/crashing by having it run in X11 mode. 

[Relevant Firefox bug report](https://bugzilla.mozilla.org/show_bug.cgi?id=1898476)

**Comment by pirosuki** [comment]
Using the experimental vulkan renderer solved the issue for me. opengl experimental seemed to crash the game even faster in my VERY limited testing. If you're unable to reach the settings menu you can try changing it in framework.ini by setting "Renderer" to "Deferred_Vulkan".
